libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -912,6 +914,105 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 			m_context << success;
 			break;
 		}
+		case FunctionType::Kind::ABIEncode:
+		case FunctionType::Kind::ABIEncodePacked:
+		case FunctionType::Kind::ABIEncodeWithSelector:
+		case FunctionType::Kind::ABIEncodeWithSignature:
+		{
+			bool const isPacked = function.kind() == FunctionType::Kind::ABIEncodePacked;
+			bool const hasSelectorOrSignature =
+				function.kind() == FunctionType::Kind::ABIEncodeWithSelector ||
+				function.kind() == FunctionType::Kind::ABIEncodeWithSignature;
+
```

Comment:
I'd add an assert here (even though the type system should validate it): `if (hasSelectorOrSignature) assert(arguments.size() >= 1)`

Reply:
Added it inside the `hasSelectorOrSignature`-if where `arguments[0]` is actually used.